articles/server-platforms/nodejs/_includes/_api_authz.md

```diff
@@ -0,0 +1,234 @@
+We will need to add a few routes for the application, including: `/` `/logout'` `/callback` and `/user`. We will also need template files for `index` and `user`.
+
+```js
+// routes/index.js
+
+router.get('/', function(req, res, next) {
+  res.render('index', { title: 'NodeJS Quickstart Demo' });
+});
+
+// Perform session logout and redirect to homepage
+router.get('/logout', function(req, res){
+  req.logout();
+  res.redirect('/');
+});
+
+// Perform the final stage of authentication and redirect to '/user'
+router.get('/callback',
+  passport.authenticate('auth0', { failureRedirect: '/url-if-something-fails' }),
+  function(req, res) {
+    res.redirect(req.session.returnTo || '/user');
+  });
+```
+
+After the authetication is complete, the `user` page should be displayed. Add a new file called `routes/user.js`.
+
+```js
+// routes/user.js
+
+var express = require('express');
+var passport = require('passport');
+var ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn();
+var router = express.Router();
+
+// Get the user profile
+router.get('/', ensureLoggedIn, function(req, res, next) {
+  res.render('user', { user: req.user });
+});
+
+module.exports = router;
+```
+
+We can now add some template files for these pages. Start with a base template called `layout.jade`. The CDN link to Auth0's JavaScript library can be placed here. Alternatively, may prefer to place this `script` tag in another layout which places it after the `head` of the page.
+
+```jade
+// views/layout.jade
+
+doctype html
+html
+  head
+    title= title
+    link(rel='stylesheet', href='/stylesheets/style.css')
+    script(src="https://cdn.auth0.com/w2/auth0-7.2.js")
+  body
+    block content
+```
+
+Next, create the `index` view. It is here where you will instantiate the Auth0 client and create a function that will initiate the authentication process when the Login button is clicked by the user.
+
+```jade
+// views/index.jade
+
+extends layout
+
+block content
+
+  h1= Welcome to my application
+  br
+  button(onclick="signIn()") Login
+
+  script.
+    // Construct an instance of the auth0 client
+    var auth0 = new Auth0({
+      domain:       '#{env.AUTH0_DOMAIN}',
+      clientID:    '#{env.AUTH0_CLIENT_ID}',
+      callbackURL: '#{env.AUTH0_CALLBACK_URL}',
+    });
+    
+    // this function will be called when a user clicks the login button
+    function signIn() {
+      auth0.login({
+      responseType: 'id_token token',
+      scope: 'openid profile {API SCOPES}',
+      audience: '{API IDENTIFIER}'
+      });
+    }
+
+```
+
+The `audience` parameter should contain your API identifier from the Dashboard. If you don't send this, the runtime will take it from the tenant settings (`tenant.default_audience` or you can set it in the Dashboard). The `scope` parameter should include one or more scopes (separated by a space) you defined in the Dashboard for your API, in addition to any of the standard [openid scopes](https://auth0.com/docs/scopes).
```

Comment:
`The scope parameter should include one or more` -> `The scope parameter may include one or more`